test/utils/matchers.go

```diff
@@ -122,16 +125,33 @@ func (m *Matcher) Eventually(obj runtime.Object, intervals ...interface{}) gomeg
 
 // eventuallyObject gets an individual object from the API server
 func (m *Matcher) eventuallyObject(obj Object, intervals ...interface{}) gomega.GomegaAsyncAssertion {
+
 	key := types.NamespacedName{
 		Name:      obj.GetName(),
 		Namespace: obj.GetNamespace(),
 	}
+
 	get := func() Object {
-		err := m.Client.Get(context.TODO(), key, obj)
+		var u Object
+		switch obj.(type) {
+		case *appsv1.StatefulSet:
+			u = &appsv1.StatefulSet{}
+		case *corev1.ConfigMap:
+			u = &corev1.ConfigMap{}
+		case *corev1.Secret:
+			u = &corev1.Secret{}
+		case *appsv1.Deployment:
+			u = &appsv1.Deployment{}
+		case *appsv1.DaemonSet:
+			u = &appsv1.DaemonSet{}
+		}
+
```

Comment:
What about using `u := u.DeepCopy()` instead, should have the same effect without having to utilise a list of known types?

Reply:
Yeah, that should work just as well.

I'll try it once I get the tests pass reliably.

Reply:
@JoelSpeed The `utils.Object` interface unfortunately doesn't have a `DeepCopy()`, so that doesn't work.
Do you feel like elaborating on your scheme suggestion before I leave it as-is?

Reply:
I thought `utils.Object` embedded `runtime.Object` which has `DeepCopy()`? Or is it `DeepCopyObject()` maybe? 🤔 I thought perhaps you could use https://godoc.org/k8s.io/apimachinery/pkg/runtime#Scheme.New 🤔